### PR TITLE
Develop render

### PR DESCRIPTION
**Pull request by a-caine**
File restructuring changes should be merged into other branches ASAP as this could cause some significant merge conflicts if left too late.
Also adds basic resource loading file which can be extended to work with other file types.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -5,8 +5,8 @@
 
 #include <cglm/cglm.h>
 
-#include "log/log.h"
-#include "log/proginfo.h"
+#include "util/log/log.h"
```
Comment (inventor02) [inline src/main.c:8]: please order alphabetically

File: src/render/render_master.c [hunk 2]
```diff
@@ -33,8 +33,10 @@ int run() {
 		clogf(CAVERNA_LOG_LEVEL_TRACE, "Element %i : Value %f\n", i, vertices[i]);
 	}
 
-	const char *vertexShaderSource = "#version 330 core\nlayout (location = 0) in vec3 aPos;\nvoid main()\n{\n   gl_Position = vec4(aPos.x, aPos.y, aPos.z, 1.0);\n}\0";
-	const char *fragmentShaderSource = "#version 330 core\nout vec4 FragColor;\nvoid main()\n{\n   FragColor = vec4(1.0f, 0.5f, 0.2f, 1.0f);\n}\0";
+	char *vertexShaderSource = load_shader_source("example.vert");
+	char *fragmentShaderSource = load_shader_source("example.frag");
+
+	printf("VERT pointer: %i, FRAG pointer: %i\n", vertexShaderSource, fragmentShaderSource);
```
Comment (inventor02) [inline src/render/render_master.c]: clog :(

File: src/util/files/resource_loader.c [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+//TODO: REMOVE THIS IN THE FUTURE, ONLY FOR TESTING
+//TODO: REPLACE ALL MALLOC WITH CALLOC
+#include <limits.h>
+
+
+#if defined(_WINDOWS) || defined(_WIN32) || defined(_WIN64)
```
Comment (inventor02) [inline src/util/files/resource_loader.c:8]: please indent appropriately

File: src/util/files/resource_loader.c [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+#include <direct.h>
+#define GetCurrentDirectory _getcwd
+#else
+#include <unistd.h>
+#define GetCurrentDirectory getcwd
```
Comment (inventor02) [inline src/util/files/resource_loader.c]: usually I would not advocate for preprocessor definitions and includes in the same block but it's kind of OK here - maybe add some comments to describe what this actually does though?

File: src/util/files/resource_loader.h [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * Contains functions to load various resources
+ * from the resource folder.
+ *
+ * @author Adam Caine
+ */
+#ifndef RESOURCE_LOADER_H
+#define RESOURCE_LOADER_H
+
+#define SHADER_SOURCE_MAX_LENGTH 1024
```
Comment (inventor02) [inline src/util/files/resource_loader.h]: preprocessor definitions should probably be below `#include`s

**Review by inventor02 — DISMISSED** [review]
Some minor points of clarification and formatting changes but once they are made the content is fine so I will approve :)

File: src/main.c (hunk 1 above)
Comment (tgodfrey0) [reply src/main.c:8]: Hi @inventor02 ,

This diff shows that the lines have been removed

Comment (inventor02) [reply src/main.c:8]: Hi @a-caine,

Apologies, there is an error in this comment.

Please ignore this comment.

With kindest regards

**Comment by a-caine** [comment]
Hi George,

I should have resolved the issues you commented on, would you mind taking another look at the pull request?

Kind regards,
Adam

**Review by tgodfrey0 — APPROVED** [review]

**Pull request merged**